src/Cookie/CookieJar.php

```diff
@@ -86,6 +86,21 @@ public static function shouldPersist(
         return false;
     }
 
+    /**
+     * Finds and returns the cookie based on the name
+     *
+     * @param string $name cookie name to search for
+     * @return SetCookie|null cookie that was found or null if not found
+     */
+    public function getCookieByName($name)
+    {
+        foreach($this->cookies as $cookie) {
+            if(strcasecmp($cookie->getName(), $name) === 0) {
```

Comment:
@sagikazarmark: Thanks for repinging me. I was out of the country. I can get to it later tonight. 😺